Adding a new package for DokuWiki Syntax

bgx4k3p opened the pull request:
Provides basic TAB-autocompletion and syntax highlighting for DokuWiki
markup language.

joshua-golub commented:
Bgx4k3p-

I am reviewing this package. Would you please add a sample DokuWiki document to your repository so that I can verify that it works as advertised?

-Josh

FichteFoll commented:
You'll need to change `"branch": "master"` to `"tags": true` and use [tags](http://git-scm.com/book/en/v2/Git-Basics-Tagging) to create releases and push valid [**semantic version**](http://semver.org) tags to your repo. As mentioned [in the docs](https://packagecontrol.io/docs/submitting_a_package#Step_4), we no longer accepts branch-based releases for new packages.


Because this is would be your first release, the version `1.0.0` could be appropriate.

```sh
$ git tag v1.0.0
$ git push origin v1.0.0
```

Package Control will strip the `v` prefix.

-

You should not include `package-metadata.json` in your repo. This file is will be automatically created by Package Control on installation.

-

`.sublime-syntax` support has been added with build 3092. Please adjust your version selector accordingly.

bgx4k3p commented:
Hello,

I removed the package-metadata.json and added tag v1.0.0, as well as a sample page per your instructions. Let me know if I need to do anything else? 

Thanks.


> On Nov 6, 2016, at 6:36 PM, FichteFoll <notifications@github.com> wrote:
> 
> You'll need to change "branch": "master" to "tags": true and use tags <http://git-scm.com/book/en/v2/Git-Basics-Tagging> to create releases and push valid semantic version <http://semver.org/> tags to your repo. As mentioned in the docs <https://packagecontrol.io/docs/submitting_a_package#Step_4>, we no longer accepts branch-based releases for new packages.
> 
> Because this is would be your first release, the version 1.0.0 could be appropriate.
> 
> $ git tag v1.0.0
> $ git push origin v1.0.0
> Package Control will strip the v prefix.
> 
> You should not include package-metadata.json in your repo. This file is will be automatically created by Package Control on installation.
> 
> .sublime-syntax support has been added with build 3092. Please adjust your version selector accordingly.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/wbond/package_control_channel/pull/5863#issuecomment-258720327>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AF1bvtFdEJXuyKcturJFDzlz3Nb8G1iTks5q7mSJgaJpZM4KRT7y>.
>

sentience commented:
@bgx4k3p Would you be interested at all in collaborating on combining your package with my old one, [here](https://github.com/sentience/DokuWiki)? We could roll out the merged package as an ST3-compatible update to that original package (enabling users of the old package to auto-update).

I think the snippets in my package (which let you tab out of, say, a heading after typing the heading text) are a bit better, but you’ve done the important work of creating a ST3-compatible syntax definition. It would be good for one package to contain the best of both worlds.

If you prefer just to go with what you’ve got, that’s okay too of course. :)

bgx4k3p commented:
@sentience Thanks for the offer, but personally I prefer my tab-completions. You can check the **DokuWiki.sublime-completions** file for all current completions. I've added some that I use very often myself, but the users can customize the file and add more. Also I like it better that it is all contained in one file and not separate snippet files. 

You can start typing **w**(as in wiki) and h1+tab for Heading 1, or use **wbash+tab** if you are about to type a bash code. The formatting works with DokuWiki's native Geshi code highlighting. ST3 is smart to autocomplete even if you haven't typed the full tag. For example, for page title, instead of doing **wtitle+tab** you can just do **wti+tab** and it works. Very quick once you use to the shortcuts.

FichteFoll commented:
> I removed the package-metadata.json and added tag v1.0.0, as well as a sample page per your instructions. Let me know if I need to do anything else?

`package-metadata.json` is still on the repo and you didn't change the package data in this PR to `"tags": true`, neither did you adjust the version selector.

bgx4k3p commented:
Sorry I’m new to GitHub and misunderstood. I believe I applied the required updates.


> On Nov 17, 2016, at 5:11 AM, FichteFoll <notifications@github.com> wrote:
> 
> I removed the package-metadata.json and added tag v1.0.0, as well as a sample page per your instructions. Let me know if I need to do anything else?
> 
> package-metadata.json is still on the repo and you didn't change the package data in this PR to "tags": true, neither did you adjust the version selector.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/wbond/package_control_channel/pull/5863#issuecomment-261207250>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AF1bvsu9p8mbIUsIX_mAvT0yc22uYe13ks5q_ChpgaJpZM4KRT7y>.
>

FichteFoll commented on repository/d.json:
```diff
@@ -938,6 +938,18 @@
+			"releases": [
+				{
+					"sublime_text": "*",
+					"version": "1.0.0",
```
This must not be here.

FichteFoll commented:
@bgx4k3p ping

FichteFoll commented:
>`.sublime-syntax` support has been added with build 3092. Please adjust your version selector accordingly.
